src/Language/PureScript/Types.hs

```diff
@@ -109,12 +107,30 @@ data Type
   | ParensInType Type
   deriving (Show, Read, Eq, Ord)
 
--- |
--- A typeclass constraint
---
-type Constraint = (Qualified (ProperName 'ClassName), [Type])
+-- | Additional data relevant to type class constraints
+data ConstraintData
+  = PartialConstraintData [[String]] Bool -- ^ a constraint generated by the exhaustivity checker
```

Comment:
(Now that I've reviewed this properly I know what they mean, but I think it might still be worth adding more information to the comments here.)